packages/settingeditor/src/SettingsFormEditor.tsx

```diff
@@ -189,6 +190,11 @@
     this.setState({ isModified: false });
   };
 
+  private _syncFormDataWithSettings = (sender: any) => {
+    this._formData = this.props.settings.composite as ReadonlyJSONObject;
+    this.setState({ isModified: this.props.settings.isModified });
```

Comment:
To address the CodeQL warning:

```suggestion
    this.setState((prevState, props) => { isModified: props.settings.isModified });
```

See https://legacy.reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous

But I have not tested this. Which leads me to - it would  be good to add a very small integration test using galata here.

Reply:
Could you confirm if the test involves the following: changing a setting (e.g., theme) and verifying that the settings editor form updates accordingly?

Reply:
Yes, that would be a good test